src/frontends/src/query_frontend.hpp

```diff
@@ -178,15 +178,15 @@ class RequestHandler {
               event_type);
           if (event_type == "set") {
             std::string model_name = key;
-            int new_version = clipper::redis::get_current_model_version(
+            std::string new_version = clipper::redis::get_current_model_version(
                 redis_connection_, key);
-            if (new_version >= 0) {
+            if (new_version.size() > 0) {
```

Comment:
If it's the empty string "" -- we could do the full check for validity, but it might be redundant with the validation in management frontend: https://github.com/ucbrise/clipper/pull/197/files#diff-9d52ffc36b86c3e051c47bd5971c718dR506